### PR TITLE
test: asyncify integration/associations

### PR DESCRIPTION
**Pull request by jedwards1211**
_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by SimonSchick** [comment]
@jedwards1211 mind sharing the tool you are using for this?

**Comment by jedwards1211** [comment]
@SimonSchick yeah, it's https://github.com/codemodsquad/asyncify

Also I'm using this temp codemod:
```js
const asyncify = require('./asyncify')

function transformer(file, api, options) {
  const j = api.jscodeshift;

  // apply main asyncify transform
  const asyncified = asyncify(file, api, options);

  // convert all return statements within it() test functions to await statements.
  const root = j(asyncified);
  root.find(j.ReturnStatement).filter(
    path => {
      if (!path.node.argument) return false
      const closestParentFn = j([path]).closest(j.Function)
      if (!closestParentFn.size()) return false
      const parentFn = closestParentFn.paths()[0]
      if (!parentFn.parentPath) return false
      const grandparent = parentFn.parentPath.node
      return grandparent.type === 'CallExpression' && grandparent.callee.type === 'Identifier' && grandparent.callee.name === 'it'
    }
  ).forEach(path => {
    j([path]).closest(j.Function).nodes()[0].async = true
    path.replace(j.expressionStatement(path.node.argument.type === 'AwaitExpression' ? path.node.argument : j.awaitExpression(path.node.argument)))
  })

  // Some old tests share values between promise handlers on a "ctx" object.
  // This replaces:
  // * `ctx.foo` -> `foo`
  // * `ctx.thing = thing` -> (removed)
  // * `ctx.thing = ...` -> `const thing = ...`
  root.find(j.VariableDeclarator, {
    id: {type: 'Identifier', name: 'ctx'},
    init: {type: 'ObjectExpression'},
  }).forEach(path => {
    const fn = j([path]).closest(j.Function)
    path.prune()

    fn.find(j.MemberExpression, {
      object: {type: 'Identifier', name: 'ctx'},
      property: {type: 'Identifier'}
    }).forEach(path => {
      const {parentPath, node} = path
      const {name} = node.property
      if (!parentPath) return
      if (parentPath.node.type === 'AssignmentExpression' && parentPath.node.left === node) {
        const statement = j([parentPath]).closest(j.Statement)
        const {right} = parentPath.node
        if (right.type !== 'Identifier' || right.name !== name) {
          statement.insertAfter([
            j.variableDeclaration('const', [
              j.variableDeclarator(
                j.identifier(name),
                right
              )
            ])
          ])
        }
        parentPath.prune()
      } else {
        path.replace(j.identifier(name))
      }
    })
  })

  return root.toSource();
}
transformer.parser = 'babylon'
module.exports = transformer
```

File: test/integration/associations/scope.test.js [hunk 1]
```diff
@@ -321,96 +315,95 @@ describe(Support.getTestDialectTeaser('associations'), () => {
...
+              this.postB.getCategories(),
+              this.postB.getTags(),
+              this.postC.getCategories(),
+              this.postC.getTags()
+            ]);
+
+            expect(postACategories.length).to.equal(1);
+            expect(postATags.length).to.equal(1);
+            expect(postBCategories.length).to.equal(1);
+            expect(postBTags.length).to.equal(1);
```
Comment (jedwards1211) [inline test/integration/associations/scope.test.js:357]: @sushantdhiman this assertion failed in an unrelated PR: https://ci.appveyor.com/project/sushantdhiman/sequelize/builds/33045550

But I can't see anything wrong with the code changes to this test...hopefully there's not anything I broke in core?  Have you been seeing failures here periodically?

Comment (sushantdhiman) [reply test/integration/associations/scope.test.js:357]: https://github.com/sequelize/sequelize/blob/e34efd477d051af6796c277406be26bb1d2c2f23/test/integration/associations/scope.test.js#L318

I have noticed this test failing a few times now as well. AFAIR only fails for MSSQL, will try to figure out what is going on if I get chance. Perhaps some promise is not returned somewhere in core codebase

Comment (jedwards1211) [reply test/integration/associations/scope.test.js:357]: Do you think it started failing after I asyncified core?

Comment (sushantdhiman) [reply test/integration/associations/scope.test.js:357]: Can't say exactly when it started failing but looks related to `bluebird removal` / `async conversion`

Comment (SimonSchick) [reply test/integration/associations/scope.test.js:357]: Didn't we have another case where the bluebird removal made code fast enough to uncover an unchained promise/race condition?

Comment (jedwards1211) [reply test/integration/associations/scope.test.js:357]: Heh, I'm kinda confused, didn't Bluebird bill itself as being the fastest promise implementation out there?  Or did that change once promises became native in V8?

Comment (jedwards1211) [reply test/integration/associations/scope.test.js:357]: Ah yup it changed by Node 12: https://blog.kuzzle.io/bluebird-vs-native-vs-async/await-state-of-promises-performances-in-2019
https://v8.dev/blog/fast-async

**Pull request merged**